### PR TITLE
DB docs for IAM authentication and revert

### PR DESCRIPTION
**Pull request by mr337**
## Description

This PR expands on the DB docs to provide information on RDS IAM and how to revert back to using DB username and pass.

## Code Review Verification Steps

* [x] Request review from a member of a different team.
* [x] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/168090813) for this change

### CODE REVIEW

File: docs/database.md [hunk 1]
```diff
@@ -28,3 +28,61 @@ When looking up objects that have a one-to-many relationship, ORMs such as Pop c
...
+
+Update the related environment configuration to match. Note that the database user is normally different than `master` as additional configuration is needed to allow a database user to login via IAM. MilMove convention for IAM enabled user is `ecs_user`. Below is a snippet of the [experimental environment config](https://github.com/transcom/mymove/blob/master/config/env/experimental.env):
+
+```ini
+DB_USER=YOUR_DB_USER_HERE OR "ecs_user"
+DB_IAM=true
+DB_IAM_ROLE=YOUR_CONTAINER_ROLE_ARN_HERE
+```
+
+### Reverting Task to use password authentication
```
Comment (mr337) [inline docs/database.md]: I would like to get a good set of 👀 on this as I don't want to experiment with environments on this. The only part I partially question is how chamber injects the `db_password` from the Param store.

File: docs/database.md [hunk 2]
```diff
@@ -28,3 +28,61 @@ When looking up objects that have a one-to-many relationship, ORMs such as Pop c
...
+
+In the event of a IAM failure it may be desired to revert back to conventional username and password authentication.
+
+1. Generate new database password
```
Comment (chrisgilmerproj) [inline docs/database.md]: Let's just move the db password into the `admin` vault in the DP3 1Password. And if they need it the procedure would be: `Get password from Infra from the admin vault in DP3 1Password`.

File: docs/database.md [hunk 3]
```diff
@@ -28,3 +28,61 @@ When looking up objects that have a one-to-many relationship, ORMs such as Pop c
...
+
+1. Update the RDS with the new password
```
Comment (chrisgilmerproj) [inline docs/database.md]: We can remove this part.

File: docs/database.md [hunk 4]
```diff
@@ -28,3 +28,61 @@ When looking up objects that have a one-to-many relationship, ORMs such as Pop c
...
+
+1. Update the Parameter store with the new password
+
+1. ```bash
+   chamber write app-YOURENV db_password NEW_PASSWORD
+   ```
+
+1. Update the environment configuration files to disable IAM authentication. Keep in mind the database user may need to change.
```
Comment (chrisgilmerproj) [inline docs/database.md]:
```suggestion
1. Update the environment configuration files to disable IAM authentication. Keep in mind the database `user` will need to be set to `master`.
```

File: docs/database.md [hunk 5]
```diff
@@ -28,3 +28,61 @@ When looking up objects that have a one-to-many relationship, ORMs such as Pop c
...
+    {
+      "name": "DB_IAM_ROLE",
+      "value": "{{ .DB_IAM_ROLE }}"
+    },
+```
+
+Update the related environment configuration to match. Note that the database user is normally different than `master` as additional configuration is needed to allow a database user to login via IAM. MilMove convention for IAM enabled user is `ecs_user`. Below is a snippet of the [experimental environment config](https://github.com/transcom/mymove/blob/master/config/env/experimental.env):
+
+```ini
+DB_USER=YOUR_DB_USER_HERE OR "ecs_user"
```
Comment (chrisgilmerproj) [inline docs/database.md]:
```suggestion
DB_USER=ecs_user
```

File: docs/database.md [hunk 6]
```diff
@@ -28,3 +28,61 @@ When looking up objects that have a one-to-many relationship, ORMs such as Pop c
...
+
+An RDS instance must be configured with IAM authentication before connecting. All RDS in the MilMove environment has this enabled. If wishing to connect to a RDS instance first prepare the infrastructure to allow this with this [guide](https://github.com/transcom/ppp-infra/blob/master/docs/runbook/0009-accessing-the-db-with-iam.md). Once complete locally milmove server can be ran with the following
+
+```bash
+/path/to/milmove serve --db-iam --db-iam-role arn:aws:iam::AWSACCOUNT:role/CONNECTROLE  --db-region us-east-2 --db-host RDSURL  --db-ssl-mode verify-full --db-ssl-root-cert bin/rds-combined-ca-bundle.pem  --db-user db_user
+```
+
+### ECS Task connecting to RDS using IAM authentication
+
+ECS tasks such as `app` and `app-client-tls` use RDS IAM authentication to securely connect without a username or passwords to rotate. This is accomplished by ECS assigning a role to the container that is allowed to connect to a specific database via IAM.
```
Comment (chrisgilmerproj) [inline docs/database.md]: I think you can just say `ECS tasks` and leave out the naming. Then we don't have to come back and update this ever.

```suggestion
ECS tasks use RDS IAM authentication to securely connect without a username or passwords to rotate. This is accomplished by ECS assigning a role to the container that is allowed to connect to a specific database via IAM.
```

File: docs/database.md [hunk 7]
```diff
@@ -28,3 +28,61 @@ When looking up objects that have a one-to-many relationship, ORMs such as Pop c
...
+
+```json
+    {
+      "name": "DB_IAM",
+      "value": "{{ .DB_IAM }}"
+    },
+    {
+      "name": "DB_IAM_ROLE",
+      "value": "{{ .DB_IAM_ROLE }}"
+    },
```
Comment (chrisgilmerproj) [inline docs/database.md]: I think I'd also include `DB_USER` and `DB_REGION` in here since those are critical to the definition.

**Comment by mr337** [comment]
Hmm, build_storybook_app has this PR hung up

**Comment by mr337** [comment]
Rebased this wondering if there is some recent storybook stuff in master I am possibly missing.

**Pull request merged**